src/inmanta/module.py

```diff
@@ -2438,14 +2438,21 @@ def get_plugin_dir(self) -> Optional[str]:
         """
         raise NotImplementedError()
 
-    def _list_python_files(self, plugin_dir: str) -> Iterator[str]:
+    def _list_python_files(self, plugin_dir: str) -> list[str]:
         """Generate a list of all python files"""
-        for file_name in glob.iglob(os.path.join(plugin_dir, "**", "*.py"), recursive=True):
-            yield file_name
+        files: Dict[str, str] = {}
 
         for file_name in glob.iglob(os.path.join(plugin_dir, "**", "*.pyc"), recursive=True):
+            # Filter out pyc files in the default cache dir. Only support our compiled pyc files.
             if "__pycache__" not in file_name:
-                yield file_name
+                files[file_name[:-3]] = file_name
+
+        for file_name in glob.iglob(os.path.join(plugin_dir, "**", "*.py"), recursive=True):
+            # store the python source file if we do not have a python file
```

Comment:
Any reason for this ordering? It seems inconsistent with the finder's behavior. This way if both files exist the compiler would use the `.py` file but we'd upload the `.pyc` file to the agent.

Reply:
I don't, that would be fine by me.

Reply:
@sanderr @bartv I don't follow this entirely, does this require further action?

Reply:
This was already handled by me

Reply:
I think so. The issue here is that there is an inconsistency between this method and the finder if both a `.py` and a `.pyc` file are present. Here we prefer the `.pyc` file while the finder does the opposite (unless Bart changed that already, I haven't checked). Bart proposed to keep the behavior here and change the behavior of the finder to match this. I'm on board with that.

Reply:
Ah, nevermind then.